<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QGroupBox" name="groupBox">
      <property name="title">
       <string>Connect to Mqtt</string>
      </property>
      <layout class="QGridLayout" name="gridLayout" columnstretch="0,0,0,0">
       <item row="1" column="0">
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>Username：</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>Certfile </string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QLineEdit" name="lineEdit_cert">
         <property name="toolTip">
          <string>Nhập đường dẫn đến file Certificate nếu có</string>
         </property>
         <property name="text">
          <string>C:/Users/Admin/Downloads/ca.crt</string>
         </property>
        </widget>
       </item>
       <item row="3" column="3">
        <widget class="QCheckBox" name="checkBox_protocol">
         <property name="text">
          <string>Protocol SSl/TLS v1.2</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Port</string>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QPushButton" name="pushButton_connect">
         <property name="toolTip">
          <string>Nhấn để kết nối</string>
         </property>
         <property name="text">
          <string>Connect</string>
         </property>
        </widget>
       </item>
       <item row="0" column="3">
        <widget class="QLineEdit" name="lineEdit_server_port">
         <property name="toolTip">
          <string>Cổng kết nối, mặc định là 8883</string>
         </property>
         <property name="whatsThis">
          <string>Cổng kết nối mặc định là 8883</string>
         </property>
         <property name="inputMask">
          <string>999999</string>
         </property>
         <property name="text">
          <string>8883</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLineEdit" name="lineEdit_server_ip">
         <property name="toolTip">
          <string>Nhập địa chỉ IP</string>
         </property>
         <property name="whatsThis">
          <string>Nhập địa chỉ IP</string>
         </property>
         <property name="text">
          <string>192.168.10.22</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLineEdit" name="lineEdit_user">
         <property name="text">
          <string>homa</string>
         </property>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QLabel" name="label_7">
         <property name="text">
          <string>Password</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label">
         <property name="frameShape">
          <enum>QFrame::NoFrame</enum>
         </property>
         <property name="text">
          <string>Broker Address</string>
         </property>
        </widget>
       </item>
       <item row="1" column="3">
        <widget class="QLineEdit" name="lineEdit_pwd">
         <property name="text">
          <string>#13@09b</string>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QLabel" name="label_8">
         <property name="text">
          <string>Design by HTien</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Subcribe</string>
       </attribute>
       <widget class="QGroupBox" name="groupBox_2">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>10</y>
          <width>782</width>
          <height>153</height>
         </rect>
        </property>
        <property name="title">
         <string>Topic</string>
        </property>
        <layout class="QFormLayout" name="formLayout_2">
         <item row="0" column="0">
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Topic:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLineEdit" name="lineEdit_topic">
           <property name="toolTip">
            <string>Nhập Topic ( Có thể nhập nhiều Topic cùng lúc)</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>QOS：</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLineEdit" name="lineEdit_qos">
           <property name="toolTip">
            <string>QOS（0，1，2）</string>
           </property>
           <property name="inputMask">
            <string>9</string>
           </property>
           <property name="text">
            <string>0</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QPushButton" name="pushButton_sub">
           <property name="toolTip">
            <string>Đăng ký Topic</string>
           </property>
           <property name="text">
            <string>Subcribe</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>Publish</string>
       </attribute>
       <widget class="QGroupBox" name="groupBox_3">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>10</y>
          <width>782</width>
          <height>153</height>
         </rect>
        </property>
        <property name="title">
         <string>Topic</string>
        </property>
        <layout class="QFormLayout" name="formLayout_3">
         <item row="0" column="0">
          <widget class="QLabel" name="label_9">
           <property name="text">
            <string>Topic:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLineEdit" name="lineEdit_topic_2">
           <property name="toolTip">
            <string>Nhập Topic để Publish</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_10">
           <property name="text">
            <string>QOS：</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLineEdit" name="lineEdit_qos_2">
           <property name="toolTip">
            <string>QOS（0，1，2）</string>
           </property>
           <property name="inputMask">
            <string>9</string>
           </property>
           <property name="text">
            <string>0</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_11">
           <property name="text">
            <string>Message:</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QLineEdit" name="lineEdit_msg_pub">
           <property name="toolTip">
            <string>Nhập nội dung tin nhắn</string>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QPushButton" name="pushButton_pub">
           <property name="toolTip">
            <string>Nhấn để Publish</string>
           </property>
           <property name="text">
            <string>Publish</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="QTextBrowser" name="textBrowser">
      <property name="toolTip">
       <string>Bảng thông báo nội dung</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>pushButton_sub</sender>
   <signal>pressed()</signal>
   <receiver>MainWindow</receiver>
   <slot>slot_sub_pressed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>69</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>16</x>
     <y>373</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_connect</sender>
   <signal>pressed()</signal>
   <receiver>MainWindow</receiver>
   <slot>slot_connect_pressed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>64</x>
     <y>119</y>
    </hint>
    <hint type="destinationlabel">
     <x>120</x>
     <y>200</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_pub</sender>
   <signal>pressed()</signal>
   <receiver>MainWindow</receiver>
   <slot>slot_pub_pressed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>68</x>
     <y>288</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>slot_connect_pressed()</slot>
  <slot>slot_sub_pressed()</slot>
  <slot>slot_pub_pressed()</slot>
 </slots>
</ui>
